source/common/network/address_impl.h

```diff
@@ -184,10 +212,10 @@ class Ipv4Instance : public InstanceBase {
     IpVersion version() const override { return IpVersion::v4; }
 
     Ipv4Helper ipv4_;
-    std::string friendly_address_;
+    mutable std::string address_string_;
```

Comment:
It's probably too early in the morning, but why are there two different members which can hold string representations of address? (Here and on line 120)

Reply:
This one is in the nested struct IpHelper to hold the string form of IP address. The other one is in IpInstance, base class of the parent classes, to represent IP and port.

Reply:
OOOOOH! IP vs IP:port. Yeah, my brain was not warmed up enough to notice that. Thanks!

Reply:
Should/could we make this one a string_view that points to the substring that holds the IP address, to avoid generating the string again and the memory to store it?

Reply:
The one holds IP + port (parent class) is created based on the one holds IP only (the child class). A reverse reference looks weird though.
And given we are lazy creating them now, duplication in the IP part isn't too bad.